statsmodels/tsa/vector_ar/tests/test_var_jmulti.py

```diff
@@ -0,0 +1,577 @@
+    for ds in datasets:
+        for dt in dt_s_list:
+            if debug_mode:
+                print("\n" + dt_s_tup_to_string(dt) + ": ", end="")
+            lags = results_ref[ds][dt]["whiteness"]["tested order"]
+
+            obtained = results_sm[ds][dt].test_whiteness_new(nlags=lags)
+            # test statistic
+            err_msg = build_err_msg(ds, dt, "WHITENESS OF RESIDUALS - "
+                                            "TEST STATISTIC")
+            desired = results_ref[ds][dt]["whiteness"]["test statistic"]
+            yield assert_allclose, obtained.test_statistic, desired, \
+                rtol, atol, False, err_msg
+            # p-value
+            err_msg = build_err_msg(ds, dt, "WHITENESS OF RESIDUALS - "
+                                            "P-VALUE")
+            desired = results_ref[ds][dt]["whiteness"]["p-value"]
+            yield assert_allclose, obtained.pvalue, desired, \
+                rtol, atol, False, err_msg
+
+            obtained = results_sm[ds][dt].test_whiteness_new(nlags=lags,
+                                                             adjusted=True)
+            # test statistic (adjusted Portmanteau test)
+            err_msg = build_err_msg(ds, dt, "WHITENESS OF RESIDUALS - "
+                                            "TEST STATISTIC (ADJUSTED TEST)")
+            desired = results_ref[ds][dt]["whiteness"]["test statistic adj."]
+            yield assert_allclose, obtained.test_statistic, desired, \
+                rtol, atol, False, err_msg
+            # p-value (adjusted Portmanteau test)
+            err_msg = build_err_msg(ds, dt, "WHITENESS OF RESIDUALS - "
+                                            "P-VALUE (ADJUSTED TEST)")
+            desired = results_ref[ds][dt]["whiteness"]["p-value adjusted"]
+            yield assert_allclose, obtained.pvalue, desired, \
+                rtol, atol, False, err_msg
+
+
+def test_exceptions():
+    if debug_mode:
+        if "exceptions" not in to_test:
+            return
+        else:
+            print("\n\nEXCEPTIONS\n", end="")
+    for ds in datasets:
+        for dt in dt_s_list:
+            if debug_mode:
+                print("\n" + dt_s_tup_to_string(dt) + ": ", end="")
+
+            # instant causality:
+            ### 0<signif<1
+            yield assert_raises, ValueError,\
+                results_sm[ds][dt].test_inst_causality, \
+                0, 0  # this means signif=0
+            ### causing must be int, str or iterable of int or str
+            yield assert_raises, TypeError,\
+                results_sm[ds][dt].test_inst_causality, [0.5]  # 0.5 not an int
+
+
+def setup():
+    datasets.append(macro)  # TODO: append more data sets for more test cases.
+
+    for ds in datasets:
+        load_data(ds, data)
+        results_ref[ds] = load_results_jmulti(ds, dt_s_list)
+        results_sm[ds] = load_results_statsmodels(ds)
+        return results_sm[ds], results_ref[ds]
```

Comment:
The yielded tests are great, and the parameterization makes perfect sense given the thorough set of test cases you've put together.

Probably easiest just to describe exactly what I was having trouble doing.  The new `test_whiteness_new` method is a near-perfect match for #3683, so I wanted to load one of the test cases/results, poke at it interactively, and confirm that your version subsumes the half-baked version I threw together.  I had hoped to copy/paste the necessary code without having to checkout and set up the branch.

If I'm the only person interested in this type of use, then don't inconvenience yourself on my account.